### PR TITLE
Classify OPDS feeds by their declared content-type kind

### PR DESCRIPTION
**Pull request by jonathangreen**
## Description

Determine an OPDS feed's type from the `kind` the server declares in the response `Content-Type` (`kind=acquisition` / `kind=navigation`) instead of relying on `opds-feed-parser`'s structural guess. `DataFetcher` uses the declared kind when present, and falls back to the parser's classification only when no kind is declared.

## Motivation and Context

`opds-feed-parser` only sees the feed body, so it classifies a document as an `AcquisitionFeed` only when **every** entry has an acquisition link, and otherwise falls back to a `NavigationFeed`. It can't see the response content type, so this is the best it can do — but it's a heuristic, and it misfires.

It misfires for valid acquisition feeds whose entries legitimately have no acquisition links. For example, when a library has no patron authentication configured, the Circulation Manager omits borrow/fulfill links (they require a patron). For a collection of licensed (non–open-access) books that leaves every entry with no acquisition link, so the feed — served with `kind=acquisition` — was classified as a navigation feed. The adapter then turned each book into a navigation link, and the client rendered large title boxes instead of book covers (`BookCover` was never reached). This is a long standing bug that has annoyed me before, but I ran into it again while working on https://github.com/ThePalaceProject/circulation/pull/3465.

The content type is the authoritative, spec-defined source of a feed's kind, and `DataFetcher` is the layer that has it. Rather than letting the parser guess and patching the result, `DataFetcher` now uses the declared kind directly (in both directions) and only falls back to the parser's classification when the response declares no kind — so the unlabeled case keeps working unchanged. `NavigationFeed` and `AcquisitionFeed` share the same shape, so re-wrapping only changes the type the adapter keys off of.

## How Has This Been Tested?

- Added unit tests in `DataFetcher-test.ts`: a feed the parser would call navigation is treated as acquisition when `kind=acquisition` is declared; a feed the parser would call acquisition is treated as navigation when `kind=navigation` is declared; and when no kind is declared the parser's classification is used unchanged (both directions).
- `tsc` + `tslint` clean; all mocha tests pass (16/16, including the 4 new).
- Verified end-to-end against a live Circulation Manager feed for a library with no patron authentication: before, the feed parsed as a `NavigationFeed` (0 books, 75 navigation links); after, it resolves to an `AcquisitionFeed` (5 lanes, 75 books, every book with a cover `imageUrl`). Confirmed visually in circulation-admin — covers now render in lanes instead of gray title boxes.

## Checklist:

- [x] I have updated the documentation accordingly. 
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/web-opds-client/src/__tests__/DataFetcher-test.ts [hunk 1]
```diff
@@ -186,4 +187,91 @@ describe("DataFetcher", () => {
       fetchMock.restore();
     });
   });
+
+  describe("applyDeclaredFeedType()", () => {
```
Comment (tdilauro) [inline packages/web-opds-client/src/__tests__/DataFetcher-test.ts:191]: It would be good to add a test for the no Content-Type case (unless there is already one somewhere else).

File: packages/web-opds-client/src/DataFetcher.ts [hunk 2]
```diff
@@ -89,6 +99,47 @@ export default class DataFetcher {
...
+    parsedData: OPDSFeed | OPDSEntry,
+    response: Response
+  ): OPDSFeed | OPDSEntry {
+    // Only feeds have a kind; entry documents are returned unchanged.
+    if (!(parsedData instanceof OPDSFeed)) {
+      return parsedData;
+    }
+
+    const contentType = response.headers.get("content-type") ?? "";
+    const kind = /kind=(acquisition|navigation)/i
```
Comment (tdilauro) [inline packages/web-opds-client/src/DataFetcher.ts]: Minor - This regex isn't anchored (with ';'), so it could theoretically occur anywhere in the content-type. That seems unlikely in practice, so I'm not too too worried about it.

**Review by tdilauro — APPROVED** [review]
Looks good!

A couple suggestions below.

File: packages/web-opds-client/src/__tests__/DataFetcher-test.ts (hunk 1 above)
Comment (jonathangreen) [reply packages/web-opds-client/src/__tests__/DataFetcher-test.ts:191]: 👍🏻 Done

File: packages/web-opds-client/src/DataFetcher.ts (hunk 2 above)
Comment (jonathangreen) [reply packages/web-opds-client/src/DataFetcher.ts]: Good idea. It prompted me to stop being lazy and trying to regex parse this and do it properly. I pulled in a dependency that does RFC content-type parsing so we handle all the edge cases properly here (quotes, malformed types, etc). I also added some test cases for some of these edge cases.
